diff --git a/sites/all/modules/contrib/og_moderation/og_moderation.module b/sites/all/modules/contrib/og_moderation/og_moderation.module
index b24dced..e52a965 100644
--- a/sites/all/modules/contrib/og_moderation/og_moderation.module
+++ b/sites/all/modules/contrib/og_moderation/og_moderation.module
@@ -163,6 +163,10 @@ function og_moderation_node_revision_access($node, $op = 'view') {
       }
     }
   }
+  // _node_revision_access() uses 'update' not 'revert'.
+  if ($op == 'revert') {
+    $op = 'update';
+  }
   return _node_revision_access($node, $op);
 }
 
